### PR TITLE
Move ROCm target status to "unmaintained".

### PR DESCRIPTION
**Pull request by stuartarchibald**
This PR:

* Removes the ROCm target code.
* Fixes up from the above.
* Remove the ROCm target docs.
* Adds a deprecation/unsupported note.
* Adds a reminder to the 0.54 CHANGE_LOG entry.

~Depends on https://github.com/numba/numba/pull/6990.~ Rebased on top of this.

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -319,7 +319,7 @@ The Numba test suite causes CI systems a lot of grief:
 #. The combination of things that Numba has to test well exceeds the capacity of
    any public CI system, (Python versions x NumPy versions x Operating systems
    x Architectures x feature libraries (e.g. SVML) x threading backends
-   (e.g. OpenMP, TBB)) and then there's CUDA and ROCm too and all their version
+   (e.g. OpenMP, TBB)) and then there's CUDA too and all its version
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst:322]: l296 there's a mention of AMD GPUs, needs removing too.

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -292,14 +292,14 @@ Platform support
...
 Every commit to the master branch is automatically tested on all of the
-platforms Numba supports. This includes ARMv7, ARMv8, POWER8, as well as both
-AMD and NVIDIA GPUs.  The build system however is internal to Anaconda, so we
-also use `Azure <https://dev.azure.com/numba/numba/_build>`_ to provide public
-continuous integration information for as many combinations as can be supported
-by the service.  Azure CI automatically tests all pull requests on Windows, OS X
-and Linux, as well as a sampling of different Python and NumPy versions. If you
-see problems on platforms you are unfamiliar with, feel free to ask for help in
-your pull request. The Numba core developers can help diagnose cross-platform
+platforms Numba supports. This includes ARMv7, ARMv8, POWER8, and NVIDIA GPUs.
```
Comment (esc) [inline docs/source/developer/contributing.rst:295]: With the advent of dropping 3.6, armv7l builds are now defunct

**Comment by esc** [comment]
Would suggest also removing the `flake8` confiugration exceptions for `roc`, now that the code is also going:

```
.flake8:    numba/roc/mathdecl.py
.flake8:    numba/roc/compiler.py
.flake8:    numba/roc/hsadecl.py
.flake8:    numba/roc/dispatch.py
.flake8:    numba/roc/hsaimpl.py
.flake8:    numba/roc/api.py
.flake8:    numba/roc/gcn_occupancy.py
.flake8:    numba/roc/stubs.py
.flake8:    numba/roc/vectorizers.py
.flake8:    numba/roc/decorators.py
.flake8:    numba/roc/hlc/config.py
.flake8:    numba/roc/hlc/common.py
.flake8:    numba/roc/hlc/hlc.py
.flake8:    numba/roc/hlc/libhlc.py
.flake8:    numba/roc/tests/hsapy/test_atomics.py
.flake8:    numba/roc/tests/hsapy/test_ufuncbuilding.py
.flake8:    numba/roc/tests/hsapy/test_autojit.py
.flake8:    numba/roc/tests/hsapy/test_intrinsics.py
.flake8:    numba/roc/tests/hsapy/test_simple.py
.flake8:    numba/roc/tests/hsapy/test_compiler.py
.flake8:    numba/roc/tests/hsapy/test_positioning.py
.flake8:    numba/roc/tests/hsapy/test_scan.py
.flake8:    numba/roc/tests/hsapy/test_reduction.py
.flake8:    numba/roc/tests/hsapy/test_memory.py
.flake8:    numba/roc/tests/hsapy/run_far_branch.py
.flake8:    numba/roc/tests/hsapy/test_math.py
.flake8:    numba/roc/tests/hsapy/test_large_code.py
.flake8:    numba/roc/tests/hsapy/test_occupancy.py
.flake8:    numba/roc/tests/hsadrv/test_driver.py
.flake8:    numba/roc/hsadrv/enums.py
.flake8:    numba/roc/hsadrv/error.py
.flake8:    numba/roc/hsadrv/enums_ext.py
.flake8:    numba/roc/hsadrv/drvapi.py
.flake8:    numba/roc/hsadrv/driver.py
.flake8:    numba/roc/hsadrv/devices.py
.flake8:    numba/roc/hsadrv/devicearray.py
.flake8:    numba/roc/servicelib/service.py
```

**Review by esc — APPROVED** [review]
I have left two comments about armv7l and flak8 excludes. Besides that it will need a conflict resolution.

**Review by esc — CHANGES_REQUESTED** [review]
Conflict resolution and removal of flake8 excludes required.

**Comment by stuartarchibald** [comment]
> Conflict resolution and removal of flake8 excludes required.

Thanks for the review. flake8 / armv7 note in 2cec26f, conflicts resolved in 69525e7.

File: docs/source/reference/deprecation.rst [hunk 3]
```diff
@@ -375,3 +375,27 @@ Schedule
...
+and support for this target in future its status will be reconsidered.
+
+Schedule
+--------
+
+In 0.54.0:
+
+- The ``ROCm`` target is officially unmaintained and the target source code has
+  been moved out of the Numba main repository and into a `separate repository
+  <https://github.com/numba/numba-rocm>`_.
```
Comment (stuartarchibald) [inline docs/source/reference/deprecation.rst:401]: NOTE: This stub needs creating.

**Comment by esc** [comment]
> > Conflict resolution and removal of flake8 excludes required.
> 
> Thanks for the review. flake8 / armv7 note in [2cec26f](https://github.com/numba/numba/commit/2cec26f6468b206c128a3d1cd5bc2185309ee55a), conflicts resolved in [69525e7](https://github.com/numba/numba/commit/69525e77b64ff812c629a8854fc9f4a4a936349d).

changes look good!

**Comment by esc** [comment]
Build Farm ID: `numba_smoketest_cpu_yaml_32`

**Comment by stuartarchibald** [comment]
> > > Conflict resolution and removal of flake8 excludes required.
> > 
> > 
> > Thanks for the review. flake8 / armv7 note in [2cec26f](https://github.com/numba/numba/commit/2cec26f6468b206c128a3d1cd5bc2185309ee55a), conflicts resolved in [69525e7](https://github.com/numba/numba/commit/69525e77b64ff812c629a8854fc9f4a4a936349d).
> 
> changes look good!

Thanks for the review.

**Review by esc — APPROVED** [review]

**Comment by esc** [comment]
The Build Farm has passed

**Comment by esc** [comment]
merge is pending conflict resolution

**Comment by stuartarchibald** [comment]
> merge is pending conflict resolution

Should be done in 32953c6, thanks.

**Comment by esc** [comment]
> > merge is pending conflict resolution
> 
> Should be done in [32953c6](https://github.com/numba/numba/commit/32953c658dbacccb0f28defc5c0faa166cc7da94), thanks.

I checked the conflict resolution locally with `gitk`, looks good!

**Review by esc — APPROVED** [review]

File: docs/source/reference/deprecation.rst (hunk 3 above)
Comment (esc) [reply docs/source/reference/deprecation.rst:454]: done: https://github.com/numba/numba-rocm

**Pull request merged**